### ISSUE TITLE
global: categorisation for admin views

### ISSUE THREAD

**Issue by lnielsen** [inveniosoftware/invenio-admin#4]
The current entry point registration does not allow to e.g. categorise related views which I think will lead to a chaotic admin. There should be a way to use the ``category``, ``menu_class_name``, ``menu_icon_type`` and ``menu_icon_value`` parameters in https://flask-admin.readthedocs.org/en/latest/api/mod_base/#flask_admin.base.BaseView

**Comment by krzysztof** [comment]
Maybe an additional kwarg dict here: https://github.com/inveniosoftware/invenio-admin/blob/master/tests/demo/onetwo.py#L56

i.e.:
``(ModelViewClass, ModelClass)`` -> ``(ModelViewClass, ModelClass, dict(category='CategoryName', menu_class_name='MyModel', ...))``

and then resolving a 3-tuple in admin init: https://github.com/inveniosoftware/invenio-admin/blob/master/invenio_admin/ext.py#L65

**Comment by krzysztof** [comment]
@lnielsen To summarize, the signature of Admin's ModelView looks like this:

http://flask-admin.readthedocs.org/en/latest/api/mod_contrib_sqla/#flask_admin.contrib.sqla.ModelView

So ModelView and the Model are actually ``args`` where the categorization and other custom view options are ``kwargs``. I see two good ways of defining an entry point - using a 3-tuple:

```Python
ep_item = (ModelView,
           Model,
           dict(category="CategoryName",
                menu_class_name="MenuClassName"),
)
```
or using a single dict:
```Python
ep_item = dict(modelview=ModelView,
               model=Model,
               category="CategoryName",
               menu_class_name="MenuClassName", ...)
```

The first one has quite explicit resolution in admin's init:

```Python
                modelview, model, kwargs = ep.load()

                # Add default security to the model view
                ProtectedAdminView = protected_adminview_factory(modelview)
                self.admin.add_view(ProtectedAdminView(model, db.session,
                    *kwargs))

```

The second requires some manual pulling of 'model' and 'modelview' from the dict and passing the rest as ``kwargs``.

I'm assuming the latter (single dict object in EP) is better ?

**Comment by lnielsen** [comment]
Fixed in #7
